LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/Processors/LnsProtocolMessageProcessor.cs

```diff
@@ -240,6 +248,7 @@ internal async Task InternalHandleDataAsync(RouteValueDictionary routeValues, We
                         }
 
                         using var scope = this.logger.BeginDeviceAddressScope(updf.DevAddr);
+                        this.uplinkMessageCounter?.Add(1);
```

Comment:
Why are all the metrics nullable? I thought we always raised them in prometheus no matter what?

Reply:
This is a bit tricky. Since we rely on the presence of the Gateway ID in the tag bag (which uses `AsyncLocal` to make it available in "ambient environment", we could potentially have interference if we raise metrics when we test. Hence we allow optional raising of metrics. Alternatively, we could also inject a `Meter` with a different namespace from the test code, if you prefer that to allow the nullability of the metrics.

Reply:
That make sense, let's continue like this